e2e/app/playwright.config.js

```diff
@@ -0,0 +1,19 @@
+import { defineConfig, devices } from '@playwright/test';
+
+import baseConfig from '../playwright.config';
+
+export default defineConfig({
+  ...baseConfig,
+  use: {
+    ...baseConfig.use,
+    baseUrl: baseConfig.use.baseUrl || "localhost:3000"
+  },
+  projects: [
+    ...baseConfig.projects,
+    // add Safari for derived app config
+    {
+      name: "webkit",
+      use: { ...devices["Desktop Safari"] },
+    },
+  ],
+});
```

Comment:
this is one of the earlier comments i gave that probably got lost, but I think we should use [a recursive/nested merge technique](https://medium.com/@abbas.ashraf19/8-best-methods-for-merging-nested-objects-in-javascript-ff3c813016d9) so that this becomes more readable. the result should be something like:

```
export default defineConfig(deepMerge(
  baseConfig,
  {
    use: {
      baseUrl: baseConfig.use.baseUrl || "localhost:3000"
    },
  }
));
```